core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -373,6 +391,14 @@
     private static final boolean VALIDATE_HEADERS = getBoolean("validateHeaders", true);
 
     static {
+        if (!isIoUringAvailable()) {
+            final Throwable cause = IOUring.unavailabilityCause();
```

Comment:
Shouldn't we access this `IOUring` class using reflection in case the dependency is not on the classpath when a user uses it?

Reply:
How about just adding it as a required dependency? It shouldn't be that large I guess. (120KB)